In [2]:
import os
print(os.getcwd())

C:\Users\HP\OneDrive\Desktop\N100_Project\notebooks


In [3]:
import os
print(os.listdir())

['.ipynb_checkpoints', 'day1_analysis.ipynb']


In [4]:
print(os.listdir("../data"))

['processed', 'raw']


In [6]:
import os

print(os.listdir("../data/raw"))

[]


In [7]:
import os

for root, dirs, files in os.walk(".."):
    for file in files:
        if file.endswith(".xlsx"):
            print(os.path.join(root, file))

In [8]:
import os

print(os.listdir("../data/raw"))

['analysis.xlsx', 'balancesheet.xlsx', 'cashflow.xlsx', 'companies.xlsx', 'documents.xlsx', 'profitandloss.xlsx', 'prosandcons.xlsx']


In [9]:
import pandas as pd

companies = pd.read_excel("../data/raw/companies.xlsx")

companies.head()

,Mkt Fintech — Nifty 100 | Companies | 92 records,Unnamed: 1,Unnamed: 2,Unnamed: 3,Unnamed: 4,Unnamed: 5,Unnamed: 6,Unnamed: 7,Unnamed: 8,Unnamed: 9,Unnamed: 10,Unnamed: 11
0,id,company_logo,company_name,chart_link,about_company,website,nse_profile,bse_profile,face_value,book_value,roce_percentage,roe_percentage
1,ABB,https://mkt.in/static/mkt-icons/nifty100/ABB.png,Abbott India Ltd,https://in.tradingview.com/chart/?symbol=NSE%3...,Abbott India Ltd is one of the leading multina...,https://www.abbott.co.in/,https://www.nseindia.com/get-quotes/equity?sym...,https://www.bseindia.com/stock-share-price/abb...,10,1657,46,34.9
2,ADANIENSOL,https://m.economictimes.com/thumb/msid-1173715...,Adani Energy Solutions Ltd,https://in.tradingview.com/chart/?symbol=NSE%3...,"AESL, part of the Adani portfolio, is a multid...",https://www.adanienergysolutions.com/,https://www.nseindia.com/get-quotes/equity?sym...,https://www.bseindia.com/stock-share-price/ada...,10,175,9,8.59
3,ADANIENT,https://mkt.in/static/mkt-icons/nifty100/ADANI...,Adani Enterprises Ltd,https://in.tradingview.com/chart/?symbol=ADANIENT,Adani Enterprises Ltd is an Indian multination...,https://www.adanienterprises.com/,https://www.nseindia.com/get-quotes/equity?sym...,https://www.bseindia.com/stock-share-price/ada...,1,363,11.6,13.64
4,ADANIGREEN,https://mkt.in/static/mkt-icons/nifty100/ADANI...,Adani Green Energy Ltd,https://in.tradingview.com/chart/?symbol=NSE%3...,"Adani Green Energy Limited, incorporated in 20...",http://www.adanigreenenergy.com/,https://www.nseindia.com/get-quotes/equity?sym...,https://www.bseindia.com/stock-share-price/ada...,10,67,96.5,14.7


In [10]:
companies.shape

(93, 12)

In [11]:
companies.columns

Index(['Mkt Fintech — Nifty 100  |  Companies  |  92 records', 'Unnamed: 1',
       'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6',
       'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11'],
      dtype='str')

In [12]:
companies.info()

<class 'pandas.DataFrame'>
RangeIndex: 93 entries, 0 to 92
Data columns (total 12 columns):
 #   Column                                                Non-Null Count  Dtype 
---  ------                                                --------------  ----- 
 0   Mkt Fintech — Nifty 100  |  Companies  |  92 records  93 non-null     str   
 1   Unnamed: 1                                            92 non-null     str   
 2   Unnamed: 2                                            93 non-null     str   
 3   Unnamed: 3                                            93 non-null     str   
 4   Unnamed: 4                                            93 non-null     str   
 5   Unnamed: 5                                            92 non-null     str   
 6   Unnamed: 6                                            92 non-null     str   
 7   Unnamed: 7                                            92 non-null     str   
 8   Unnamed: 8                                            92 non-null     object
 9   U

In [13]:
companies.isnull().sum()

Mkt Fintech — Nifty 100  |  Companies  |  92 records    0
Unnamed: 1                                              1
Unnamed: 2                                              0
Unnamed: 3                                              0
Unnamed: 4                                              0
Unnamed: 5                                              1
Unnamed: 6                                              1
Unnamed: 7                                              1
Unnamed: 8                                              1
Unnamed: 9                                              1
Unnamed: 10                                             1
Unnamed: 11                                             2
dtype: int64

In [14]:
companies.describe()

,Mkt Fintech — Nifty 100 | Companies | 92 records,Unnamed: 1,Unnamed: 2,Unnamed: 3,Unnamed: 4,Unnamed: 5,Unnamed: 6,Unnamed: 7,Unnamed: 8,Unnamed: 9,Unnamed: 10,Unnamed: 11
count,93,92,93,93,93,92,92,92,92,92,92.0,91.0
unique,93,91,93,55,93,91,92,92,5,85,85.0,86.0
top,id,https://mkt.in/static/mkt-icons/nifty100/HEROM...,company_name,https://in.tradingview.com/chart/qGsydD2w/?sym...,about_company,https://www.bajajfinserv.in/,nse_profile,bse_profile,10,275,13.2,57.1
freq,1,2,1,39,1,2,1,1,31,3,3.0,2.0


In [15]:
companies.describe(include="all")

,Mkt Fintech — Nifty 100 | Companies | 92 records,Unnamed: 1,Unnamed: 2,Unnamed: 3,Unnamed: 4,Unnamed: 5,Unnamed: 6,Unnamed: 7,Unnamed: 8,Unnamed: 9,Unnamed: 10,Unnamed: 11
count,93,92,93,93,93,92,92,92,92,92,92.0,91.0
unique,93,91,93,55,93,91,92,92,5,85,85.0,86.0
top,id,https://mkt.in/static/mkt-icons/nifty100/HEROM...,company_name,https://in.tradingview.com/chart/qGsydD2w/?sym...,about_company,https://www.bajajfinserv.in/,nse_profile,bse_profile,10,275,13.2,57.1
freq,1,2,1,39,1,2,1,1,31,3,3.0,2.0


In [16]:
analysis = pd.read_excel("../data/raw/analysis.xlsx")
balancesheet = pd.read_excel("../data/raw/balancesheet.xlsx")
cashflow = pd.read_excel("../data/raw/cashflow.xlsx")
profitandloss = pd.read_excel("../data/raw/profitandloss.xlsx")
documents = pd.read_excel("../data/raw/documents.xlsx")
prosandcons = pd.read_excel("../data/raw/prosandcons.xlsx")

In [17]:
datasets = {
    "companies": companies,
    "analysis": analysis,
    "balancesheet": balancesheet,
    "cashflow": cashflow,
    "profitandloss": profitandloss,
    "documents": documents,
    "prosandcons": prosandcons
}

for name, df in datasets.items():
    print(f"{name}: {df.shape}")

companies: (93, 12)
analysis: (21, 6)
balancesheet: (1313, 13)
cashflow: (1188, 7)
profitandloss: (1277, 15)
documents: (1586, 4)
prosandcons: (17, 4)


# Day 1 Summary

## Tasks Completed

- Project structure created
- Datasets loaded successfully
- Data shape checked
- Data types analyzed
- Missing values checked
- Basic exploratory analysis performed

## Dataset Summary

| Dataset | Rows | Columns |
|----------|------|----------|
| Companies | 93 | 12 |
| Analysis | 21 | 6 |
| Balance Sheet | 1313 | 13 |
| Cash Flow | 1188 | 7 |
| Profit & Loss | 1277 | 15 |
| Documents | 1586 | 4 |
| Pros & Cons | 17 | 4 |

## Status

Day 1 completed successfully.